server/src/main/java/org/hyperskill/community/flashcards/card/CardService.java

```diff
@@ -40,27 +39,31 @@ public class CardService {
     private final CardReadConverter converter;
     private final CardMapper mapper;
 
-    public Page<Card> getCardsByCategory(String username, String categoryId, int page) {
+    public Page<Card> getCardsByCategory(String username, String categoryId, int page, String titleFilter) {
         Objects.requireNonNull(categoryId, "Category ID cannot be null");
 
         final var category = categoryService.findById(username, categoryId);
-
-        var aggregation = Aggregation.newAggregation(
-                Aggregation.sort(Sort.by("name")),
-                Aggregation.skip((long) page * PAGE_SIZE),
-                Aggregation.limit(PAGE_SIZE)
-        );
-
-        var count = mongoTemplate.count(new Query(), category.name());
-        var result = mongoTemplate.aggregate(aggregation, category.name(), Document.class)
-                .getRawResults()
-                .getList("results", Document.class);
-        var cards = result.stream().map(converter::convert).toList();
+        var pageRequest = PageRequest.of(page, PAGE_SIZE, Sort.by("title"));
+        var query = createFilterQuery(titleFilter);
+        var count = mongoTemplate.count(query, category.name());
```

Comment:
the count query is now fixed, right?

Reply:
yep. it works now as expected.
Though on still needs care. I need to keep a strict order:
1. create query with applied filters
2. count with this query
3. use query.with(pageRequest) as the latter modifies the query - and then count always gives just the page size ...